liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java

```diff
@@ -385,10 +383,9 @@ protected Set<String> getAppConfigLocationsFromSourceServerXml() {
         return scd != null ? scd.getLocations() : new HashSet<String>();
     }
 
-    protected ServerConfigDocument getServerConfigDocument(CommonLoggerI log, File serverXML, File configDir, File bootstrapFile,
-            Map<String, String> bootstrapProp, File serverEnvFile, boolean giveConfigDirPrecedence, Map<String, File> libertyDirPropertyFiles) throws IOException {
-        if (scd == null || !scd.getServerXML().getCanonicalPath().equals(serverXML.getCanonicalPath())) {
-            scd = new ServerConfigDocument(log, serverXML, configDir, bootstrapFile, bootstrapProp, serverEnvFile, giveConfigDirPrecedence, libertyDirPropertyFiles);
+    protected ServerConfigDocument getServerConfigDocument(CommonLoggerI log, Map<String, File> libertyDirPropertyFiles) {
+        if (scd == null) {
```

Comment:
Since we cannot check for a change in serverXML file anymore, I wonder if we should just always create a new ServerConfigDocument here?

Reply:
I think we should keep this singleton condition. I can see that in some places we are setting scd=null for force reinitialization

![image](https://github.com/user-attachments/assets/25f820b8-c0bb-427d-b9df-b4f908aa5f0e)